core/src/main/scala/com/permutive/refreshable/Refreshable.scala

```diff
@@ -53,196 +55,73 @@ trait Refreshable[F[_], A] { outer =>
   def restart: F[Boolean]
 
   def map[B](f: A => B): Refreshable[F, B] = new Refreshable[F, B] {
-    implicit override protected def functor: Functor[F] = outer.functor
-    override def get: F[CachedValue[B]] = outer.get.map(_.map(f))
-    override def cancel: F[Boolean] = outer.cancel
-    override def restart: F[Boolean] = outer.restart
+    implicit override protected def functor: Functor[F] = self.functor
+    override def get: F[CachedValue[B]] = self.get.map(_.map(f))
+    override def cancel: F[Boolean] = self.cancel
+    override def restart: F[Boolean] = self.restart
   }
 
   def mapK[G[_]: Functor](fk: F ~> G): Refreshable[G, A] =
-    Refreshable.mapK(this)(fk)
+    new Refreshable[G, A] {
+      override protected def functor: Functor[G] = implicitly
+
+      override def get: G[CachedValue[A]] = fk(self.get)
+      override def cancel: G[Boolean] = fk(self.cancel)
+      override def restart: G[Boolean] = fk(self.restart)
+    }
+
 }
 
-object Refreshable {
+trait RefreshableUpdates[F[_], A] extends Refreshable[F, A] { self =>
```

Comment:
I kind of like putting the subclass inside the object so we have `Refreshable.Updates`.

Reply:
sure 👍